internal/pkg/ssm/command.go

```diff
@@ -94,7 +98,7 @@ func RunCommand(session *session.Session, logger logr.Logger, instanceId, comman
 
 	logger.Info("Waiting for ssm command to finish")
 	var commandOut *ssm.GetCommandInvocationOutput
-	r := retrier.New(300*time.Minute, retrier.WithMaxRetries(2160, 60*time.Second))
+	r := retrier.New(timeout, retrier.WithMaxRetries(2160, 60*time.Second))
```

Comment:
I would make this one slightly bigger
```suggestion
    // Making the retrier wait for longer than the provided SSM timeout to make sure
    // we always get the output results.
	r := retrier.New(timeout + 5 * time.Minute, retrier.WithMaxRetries(2160, 60*time.Second))
```

Now that I think about it, maybe we need to change the option as well

```suggestion
	// Making the retrier wait for longer than the provided SSM timeout to make sure
    // we always get the output results.
	r := retrier.New(timeout + 5 * time.Minute, retrier.WithMaxRetries(math.MaxInt, 60*time.Second))
```
That should guarantee "infinite" (not really) retries, so we only error out if max timeout is reached.